python/ray/autoscaler/updater.py

```diff
@@ -181,10 +181,12 @@ def do_sync(remote_path, local_path, allow_non_existing_paths=False):
 
             with LogTimer(self.log_prefix +
                           "Synced {} to {}".format(local_path, remote_path)):
-                self.cmd_runner.run(
-                    "mkdir -p {}".format(os.path.dirname(remote_path)),
-                    run_env="host")
-                sync_cmd(local_path, remote_path)
+                if not isinstance(self.cmd_runner, DockerCommandRunner):
+                    # The DockerCommandRunner handles this internally
+                    self.cmd_runner.run(
+                        "mkdir -p {}".format(os.path.dirname(remote_path)),
+                        run_env="host")
```

Comment:
it's not harmful to keep this here right?

Reply:
It can be--the main reason is that sometimes the mount might be to `/root` in the container. Trying to `mkdir` to such a path on the host would result in a permission error.